### ISSUE TITLE
Config errors when generating a new config file on a new world.

### ISSUE THREAD

**Issue by BastionNtB** [Cazsius/Spice-of-Life-Carrot-Edition#113]
Just an FYI, it seems to generate the file correctly, but seems the values it tries to create it with aren't valid, and some are improper. Generating lots of errors. This error was generated on the client during a local world creation, but I'm pretty sure this happens on a server world creation as well.

 --version, 1.16.5-forge-36.1.23
solcarrot-1.16.3-1.9.10.jar

[01Jun2021 16:30:18.944] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file [PATH]\adventurepack\1.16.5\client\saves\New World\serverconfig\solcarrot-server.toml is not correct. Correcting
[01Jun2021 16:30:18.945] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key milestones was corrected from null to its default, SimpleCommentedConfig:{}. 
[01Jun2021 16:30:18.946] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key milestones.baseHearts was corrected from null to its default, 10. 
[01Jun2021 16:30:18.946] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key milestones.heartsPerMilestone was corrected from null to its default, 2. 
[01Jun2021 16:30:18.946] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key milestones.milestones was corrected from null to its default, [5, 10, 15, 20, 25]. 
[01Jun2021 16:30:18.947] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key filtering was corrected from null to its default, SimpleCommentedConfig:{}. 
[01Jun2021 16:30:18.947] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key filtering.blacklist was corrected from null to its default, []. 
[01Jun2021 16:30:18.947] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key filtering.whitelist was corrected from null to its default, []. 
[01Jun2021 16:30:18.947] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key filtering.minimumFoodValue was corrected from null to its default, 1. 
[01Jun2021 16:30:18.948] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key miscellaneous was corrected from null to its default, SimpleCommentedConfig:{}. 
[01Jun2021 16:30:18.948] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key miscellaneous.resetOnDeath was corrected from null to its default, false. 
[01Jun2021 16:30:18.948] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key miscellaneous.limitProgressionToSurvival was corrected from null to its default, false. 
[01Jun2021 16:30:18.950] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file [PATH]\ntb\adventurepack\1.16.5\client\saves\New World\serverconfig\servertabinfo-server.toml is not correct. Correcting
[01Jun2021 16:30:18.951] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key general was corrected from null to its default, SimpleCommentedConfig:{}. 
[01Jun2021 16:30:18.951] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key general.refreshTicks was corrected from null to its default, 100. 
[01Jun2021 16:30:18.951] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key general.opOnlyMode was corrected from null to its default, false. 
[01Jun2021 16:30:18.953] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file [PATH]\ntb\adventurepack\1.16.5\client\saves\New World\serverconfig\forge-server.toml is not correct. Correcting
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server was corrected from null to its default, SimpleCommentedConfig:{}. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.removeErroringEntities was corrected from null to its default, false. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.removeErroringTileEntities was corrected from null to its default, false. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.fullBoundingBoxLadders was corrected from null to its default, false. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.zombieBaseSummonChance was corrected from null to its default, 0.1. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.zombieBabyChance was corrected from null to its default, 0.05. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.logCascadingWorldGeneration was corrected from null to its default, true. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.fixVanillaCascading was corrected from null to its default, false. 
[01Jun2021 16:30:18.954] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.dimensionUnloadQueueDelay was corrected from null to its default, 0. 
[01Jun2021 16:30:18.955] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.treatEmptyTagsAsAir was corrected from null to its default, false. 
[01Jun2021 16:30:18.955] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Incorrect key server.fixAdvancementLoading was corrected from null to its default, true.

**Comment by juliand665** [comment]
Thanks for the report! So this happens without configuring anything? Does it still happen with just Spice of Life and no other mods?

Is it possible you have a file in the `defaultconfigs` folder that it's copying over automatically?

**Comment by BastionNtB** [comment]
I do have other mods

Here is the list:

AI Improvements
Bed Benefits
Bookshelf
Chicken Drop Feathers
Clumps
ConnectedTexturesMod
FastFurnace
FastWorkbench
FTB Backups
Game Stages
IguanaTweaks Reborn
InsaneLib
LevelHearts
PackMode
Passthrough Signs
Placebo
Polymorph
RandomPatches
ReAuth
RecipeBuffers
Screenshot to Clipboard (Fo...
Smooth Boot (Forge)
Sound Device Options
Spawner Bug Fix
Spice of Life: Carrot Edition
SwingThroughGrass
VanillaFix

However, it should be noted that this happened as a stand alone mod, as the errors would indicate, and I do not have anything in the defaultconfig folder. Nor do I have that mod :)

Perhaps that is the issue? Is the mod assuming I have a default config in that folder? Is that why the key is null?

**Comment by juliand665** [comment]
Oh, the defaultconfig folder is a forge thing actually. And no, it shouldn't rely on that at all—when I'm testing, I don't use it. I guess what I'm asking is if, rather than me installing all those mods, you could try and see if launching your game with just SoL: Carrot and creating a new world still exhibits this issue. Basically, I'd like to know if it's a conflict/problem with another mod or just our own fault, lol.

**Comment by BastionNtB** [comment]
My apologies I wasn't clear enough.

Initial tests were done with those mods listed, with no existing config.

After you mentioned, I did another test with just SoL and with no other mods, no config, but the same errors showed again.

Each test is done with creating a new world from scratch and having no configurations remaining from the previous world.

**Comment by juliand665** [comment]
Ah oof, that's interesting. I'll have to check if maybe that version of forge broke something. To be clear, the log is showing these errors, but functionally everything works fine? What if you adjust the config, when you're in the world or when you're not (it should live reload)?

**Comment by BastionNtB** [comment]
Yeah, this report was just an FYI, after it generates the file, all is well.

Just generating the file this happens. Doesn't even seem to cause issues on first load either as the settings that result as a default seem to load.

**Comment by juliand665** [comment]
Hm, I see. I'm gonna tentatively leave this open but do nothing about it, because I can't be bothered to fix my dev environment right now, lmao.

**Comment by Mattsingen** [comment]
I'm not a modder at all so I'm guessing here, but as the mod don't have a server config file on install it's created on world creation. I've tried manually adding a server config in the instance config folder and that isn't copied over when creating a world either. 
I'm guessing that this is also the problem in #118

**Comment by juliand665** [comment]
If you want to provide a default serverconfig, you need to put it in the defaultconfigs folder rather than the config folder.

My theory about #118 is that they expected to see the config file appear in the config folder, as they're not used to serverconfigs.
